MonoGame.Framework/Platform/Native/Interop.cs

```diff
@@ -0,0 +1,171 @@
+    public static partial void GraphicsDevice_BindConstantBuffer(MGG_GraphicsDevice* device, ShaderStage stage, int slot, MGG_Buffer* buffer);
+
+    #endregion
+
+    #region Buffer
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Buffer_Create", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial MGG_Buffer* Buffer_Create(MGG_GraphicsDevice* device, int length);
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Buffer_SetData", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Buffer_SetData(
+        MGG_GraphicsDevice* game,
+        ref MGG_Buffer* buffer,
+        int offset,
+        byte* data,
+        int length,
+        [MarshalAs(UnmanagedType.U1)]
+        bool discard);
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Buffer_Destroy", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Buffer_Destroy(MGG_GraphicsDevice* device, MGG_Buffer* buffer);
+
+    #endregion
+
+    #region Texture
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Texture_Create", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial MGG_Texture* Texture_Create(MGG_GraphicsDevice* device, TextureType type, SurfaceFormat format, int width, int height, int mipmaps, int slices);
+
+    [LibraryImport("monogame", EntryPoint = "MGG_RenderTarget_Create", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial MGG_Texture* RenderTarget_Create(
+        MGG_GraphicsDevice* device,
+        TextureType type,
+        SurfaceFormat format,
+        int width,
+        int height,
+        int mipmaps,
+        int slices,
+        DepthFormat depthFormat,
+        int multiSampleCount,
+        RenderTargetUsage usage);
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Texture_Destroy", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Texture_Destroy(MGG_GraphicsDevice* device, MGG_Texture* texture);
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Texture_SetData", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Texture_SetData(
+        MGG_GraphicsDevice* device,
+        MGG_Texture* texture,
+        int level,
+        int slice,
+        int x,
+        int y,
+        int width,
+        int height,
+        byte* data,
+        int dataBytes);
+
+    [LibraryImport("monogame", EntryPoint = "MGG_Texture_GetData", StringMarshalling = StringMarshalling.Utf8)]
+    public static partial void Texture_GetData(
+        MGG_GraphicsDevice* device,
+        MGG_Texture* texture,
+        int level,
+        int slice,
+        int x,
+        int y,
+        int width,
+        int height,
+        byte* data,
+        int dataBytes);
+
+    #endregion
+}
+
+
+/// <summary>
+/// MonoGame image native calls.
+/// </summary>
+internal static unsafe partial class MGI
```

Comment:
I think we should ditch StbSharp for native stb directly since we have a native library.

This API exposes reading images (jpg/png) from memory as well as writing jpg and png.

We could implement more here in the future.